src/lib.rs

```diff
@@ -1,268 +1,211 @@
 /*!
-Input manager for [Bevy](https://bevyengine.org), inspired by
-[Unreal Engine Enhanced Input](https://dev.epicgames.com/documentation/en-us/unreal-engine/enhanced-input-in-unreal-engine).
+A powerful observer-based input manager for Bevy.
 
-# Quick start
+The design of this crate is heavily inspired by
+[Unreal Engine Enhanced Input](https://dev.epicgames.com/documentation/en-us/unreal-engine/enhanced-input-in-unreal-engine),
+but adapted to Bevy's ECS architecture and idioms.
 
-## Prelude
+# Core Concepts
 
-We provide a [`prelude`] module, which exports most of the typically used traits and types.
-
-## Plugins
-
-Add [`EnhancedInputPlugin`] to your app:
-
-```
-use bevy::prelude::*;
-use bevy_enhanced_input::prelude::*;
-
-let mut app = App::new();
-app.add_plugins((MinimalPlugins, EnhancedInputPlugin));
-```
-
-## Core Concepts
+This crate introduces three main concepts:
 
 - **Actions** represent something a player can do, like "Jump", "Movement", or "Open Menu". They are not tied to specific input.
 - **Bindings** connect those actions to real input sources such as keyboard keys, mouse buttons, gamepad axes, etc.
 - **Contexts** represent a certain input state the player can be in, such as "On foot" or "In car". They associate actions with
   entities and define when those actions are evaluated.
 
-Contexts are regular components. Depending on your type of game, you may have a single global context
-or multiple contexts for different gameplay states. Contexts can be layered, and any number of them can be active at the same time.
-To register a component as an input context, you need to call [`InputContextAppExt::add_input_context`]. By default, contexts are
-evaluated during [`PreUpdate`], but you can customize this by using [`InputContextAppExt::add_input_context_to`] instead.
+In short, actions are mapped to inputs via bindings, and contexts control which actions are active.
 
-Actions are represented by entities with the [`Action<A>`] component, where `A` is a user-defined marker that implements the
-[`InputAction`] trait, which defines [`InputAction::Output`] type - the value the action produces. It could be [`bool`], [`f32`],
-[`Vec2`] or [`Vec3`]. Actions associated with contexts via [`ActionOf`] relationship. We provide the [`actions!`] macro, which is
-similar to [`related!`], but for actions. The relationship is generic over `C` because a single entity can have multiple associated
-contexts.
+## Actions
```

Comment:
Have you considered swapping actions and bindings to go from the bottom of the hierarchy? Or do you think this reads better when actions come first? Just asking :slightly_smiling_face:

Reply:
Bindings only make sense as a link between inputs and actions. If users don't know what actions are, they'll be lost!